examples/grammars/csv/csv.js

```diff
@@ -0,0 +1,96 @@
+"use strict"
+/**
+ * An Example of implementing a CSV Grammar with Chevrotain.
+ *
+ * Based on: https://github.com/antlr/grammars-v4/blob/master/csv/CSV.g4
+ *
+ * Note that this is a pure grammar without any actions (either embedded or via a CST Visitor).
+ */
+
+var chevrotain = require("chevrotain")
+
+// ----------------- lexer -----------------
+var createToken = chevrotain.createToken
+var Lexer = chevrotain.Lexer
+var Parser = chevrotain.Parser
+
+// Lexer
+const Text = createToken({name: "Text", pattern: /[^,\n\r"]+/})
+const Comma = createToken({name: "Comma", pattern: /,/})
+const NewLine = createToken({name: "NewLine", pattern: /\r?\n/})
+const String = createToken({name: "String", pattern: /"(?:""|[^"])*"/})
+
+const allTokens = [Text, String, Comma, NewLine]
+const CsvLexer = new Lexer(allTokens)
+
+
+// Parser
+class CsvParser extends Parser {
+
+    constructor(input) {
+        super(input, allTokens, {
+            // uncomment this to enable automatic CstOutput.
+            // outputCst: true
+        })
+
+        // not mandatory, using $ (or any other sign) to reduce verbosity (this. this. this. this. .......)
+        const $ = this
+
+        $.RULE("csvFile", () => {
+            $.SUBRULE($.hdr)
+            $.AT_LEAST_ONE(() => {
+                $.SUBRULE2($.row)
+            })
+        })
+
+        $.RULE("hdr", () => {
+            $.SUBRULE($.row)
+        })
+
+        // the parsing methods
+        $.RULE("row", () => { // using ES2015 Arrow functions to reduce verbosity.
+            $.SUBRULE($.field)
+            $.MANY(() => {
+                $.CONSUME(Comma)
+                $.SUBRULE2($.field)
+            })
+            $.CONSUME(NewLine)
```

Comment:
Yeah, that solution looks kinda obvious. Thanks.
Still, you implemented it differently: `(field (comma field)* newline)+` which is what raised an eyebrow.
In `csvFile` you have `hdr` (== `row`), and then `row?` which includes `\r\n` at the end.